src/components/Nav/Search.tsx

```diff
@@ -1,23 +1,35 @@
 import { SearchIcon } from "lucide-react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { useState } from "react";
 
 export default function Search() {
+  const [searchTerm, setSearchTerm] = useState('');
+  const router = useRouter();
+
+  const handleSearch = (e: { preventDefault: () => void; }) => {
+    e.preventDefault();
+    router.push(`/search?query=${encodeURIComponent(searchTerm)}`);
+  };
+
   return (
     <div>
       <search>
-        <form className="text-base text-base-content">
+        <form onSubmit={handleSearch} className="text-base text-base-content">
           <label htmlFor="search" className="sr-only">
             Search
           </label>
           <div className="join">
-            <div className="join-item btn rounded-l-full"><SearchIcon size={20} /></div>
+            <Link href='' onClick={handleSearch} className="join-item btn rounded-l-full"><SearchIcon size={20} /></Link>
```

Comment:
this should also probably just be a button

Reply:
that makes sense lol, long day